ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/SearchIndexTool.java

```diff
@@ -122,21 +131,36 @@ private static Map<String, Object> processResponse(SearchHit hit) {
     public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
         try {
             String input = parameters.get(INPUT_FIELD);
-            JsonObject jsonObject = GSON.fromJson(input, JsonObject.class);
-            String index = Optional
-                .ofNullable(jsonObject)
-                .map(x -> x.get(INDEX_FIELD))
-                .map(JsonElement::getAsString)
-                .orElse(parameters.getOrDefault(INDEX_FIELD, null));
-            String query = Optional
-                .ofNullable(jsonObject)
-                .map(x -> x.get(QUERY_FIELD))
-                .map(JsonElement::toString)
-                .orElse(parameters.getOrDefault(QUERY_FIELD, null));
-            if (index == null || query == null) {
-                listener.onFailure(new IllegalArgumentException("SearchIndexTool's two parameter: index and query are required!"));
+            String index = null;
+            String query = null;
+            if (input != null && !input.isEmpty()) {
```

Comment:
[nit] we can change it to : `!StringUtils.isEmpty(input)`

Reply:
Done